geocoder_tester/base.py

```diff
@@ -147,11 +147,12 @@ def assert_expected(expected):
         for r in results['features']:
             passed = True
             properties = None
-            failed = r['properties']['failed'] = []
             if 'geocoding' in r['properties']:
                 properties = r['properties']['geocoding']
+                failed = r['properties']['geocoding']['failed'] = []
```

Comment:
why do you need this ?

Reply:
The failed keys (the non matching ones) are stored and displayed with some special highlight to ease the debug. See https://github.com/geocoders/geocoder-tester/blob/master/geocoder_tester/base.py#L210